src/ray/core_worker/core_worker.h

```diff
@@ -1065,8 +1055,15 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
   // Queue of tasks to resubmit when the specified time passes.
   std::deque<std::pair<int64_t, TaskSpecification>> to_resubmit_ GUARDED_BY(mutex_);
 
-  // Plasma Callback
-  PlasmaSubscriptionCallback plasma_done_callback_;
+  // Guard for `async_plasma_callbacks_` map.
+  mutable absl::Mutex plasma_mutex_;
+
+  // Callbacks for when when a plasma object becomes ready.
+  absl::flat_hash_map<ObjectID, std::vector<std::function<void(void)>>>
+      async_plasma_callbacks_ GUARDED_BY(plasma_mutex_);
+
+  void PlasmaCallback(SetResultCallback success, std::shared_ptr<RayObject> ray_object,
```

Comment:
PlasmaCallback needs a comment

Reply:
Just to confirm, this handler is for GetAsync falling back to plasma?

Reply:
Yep!